src/model/keychain.js

```diff
@@ -136,3 +143,86 @@ export async function hasKey(key) {
   }
   return null;
 }
+
+export async function restoreBackupIntoKeychain(backedUpData) {
+  // Access control config per each type of key
+  const publicAccessControlOptions = {
+    accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY,
+  };
+
+  let privateAccessControlOptions = {};
+  const canAuthenticate = await canImplyAuthentication({
+    authenticationType: AUTHENTICATION_TYPE.DEVICE_PASSCODE_OR_BIOMETRICS,
+  });
+
+  let isSimulator = false;
+
+  if (canAuthenticate) {
+    isSimulator = __DEV__ && (await DeviceInfo.isEmulator());
+  }
+  if (canAuthenticate && !isSimulator) {
+    privateAccessControlOptions = {
+      accessControl: ACCESS_CONTROL.USER_PRESENCE,
+      accessible: ACCESSIBLE.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
+    };
+  }
+
+  Object.keys(backedUpData).forEach(async key => {
+    const value = backedUpData[key];
+    let accessControl = publicAccessControlOptions;
+    if (
+      (key.indexOf('rainbowSeedPhrase') !== -1 ||
+        key.indexOf('rainbowPrivateKey') !== -1) &&
+      key !== 'rainbowSeedPhraseMigratedKey'
+    ) {
+      accessControl = privateAccessControlOptions;
+    }
+    if (typeof value === 'string') {
+      await saveString(key, value, accessControl);
+    } else {
+      await saveObject(key, value, accessControl);
+    }
+  });
+
+  // Save the migration flag
+  // to prevent this flow in the future
+  await saveString(
+    'rainbowSeedPhraseMigratedKey',
```

Comment:
also, did we mean for this to be the oldSeedPhraseMigratedKey (which is the new key)